src/gfx/shaders/Uber_frag.glsl

```diff
@@ -287,151 +398,39 @@ varying vec3 vViewPosition;
     ReflectedLight reflectedLight = ReflectedLight( vec3( 0.0 ), vec3( 0.0 ), vec3( 0.0 ));
     vec3 irradiance = ambientLightColor * PI;
 
+    float shadowMask = 1.0;
     // use loop for number
     #if NUM_DIR_LIGHTS > 1
       for (int i = 0; i < NUM_DIR_LIGHTS; i ++ ) {
-        RE_Direct_BlinnPhong(directionalLights[i], geometry, material, reflectedLight);
+        #if defined(SHADOWMAP)
+          if(directionalLights[i].shadow > 0) {
+            shadowMask = getShadow( directionalShadowMap[ i ], directionalLights[ i ].shadowMapSize, directionalLights[ i ].shadowBias, directionalLights[ i ].shadowRadius, vDirectionalShadowCoord[ i ] );
```

Comment:
Please, optimize parameters of _getShadow_ function. For example, you put directionalLight[i] itself instead of directionalLight[i].shadow... three times. Also could remove unnecessary spaces in _[ x _] constructions.

Reply:
done